File: src/main/java/org/jukeboxmc/block/BlockLightBlock.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemLightBlock;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockLightBlock extends Block {

    public BlockLightBlock() {
        super( "minecraft:light_block" );
    }

    @Override
    public ItemLightBlock toItem() {
        return new ItemLightBlock();
    }

    @Override
    public BlockType getType() {
        return BlockType.LIGHT_BLOCK;
    }

    public void setBlockLightLevel( int value ) { //0-15
        this.setState( "block_light_level", value );
    }

    public int getBlockLightLevel() {
        return this.stateExists( "block_light_level" ) ? this.getIntState( "block_light_level" ) : 0;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockLightBlock.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.LightBlock
import org.jukeboxmc.server.block.JukeboxBlock

class BlockLightBlock(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), LightBlock {

    override fun canPassThrough(): Boolean {
        return true
    }

   override fun getBlockLightLevel(): Int {
       return this.getIntState("block_light_level")
   }

   override fun setBlockLightLevel(value: Int): LightBlock {
       return this.setState("block_light_level", value)
   }
}